R/regression.R

```diff
@@ -1912,6 +1960,35 @@ updateStackedFormula <- function(data, formula)
     return(new.formula)
 }
 
+#' Throw warning to the user that there are some codes used in a data reduction that are not seen
+#' elsewhere in the codeframe, e.g. a NET has code A and B and the code A is seen in the codeframe
+#' but the code B is not seen elsewhere in the codeframe. So removing the data reduction will lose
+#' the information about code B.
+#' @param reduction.list List that contains two elements, \itemize{
+#'  \item nets Character vector of net(s) that exhibit this situation
+#'  \item variable.type Character vector stating the variable set structure that this applies to.
+#' e.g. The predictor grid or outcome variables. Could be a single element or vector with two elements
+#' if both outcome and predictor grids are affected.
+#' }
+#' @noRd
+throwCodeReductionWarning <- function(reduction.list)
+{
+    nets <- reduction.list[["nets"]]
+    variable.type <- reduction.list[["variable.type"]]
+    net.name.txt <- paste0(sQuote(nets, q = FALSE))
+    if (length(nets) > 1)
+        net.name.txt <- paste0(": (", paste0(net.name.txt, collapse = "; "), ")")
+    net.txt <- ngettext(length(nets), "a NET, ", "NETs")
+    variable.type.and <- paste0(paste0(variable.type, collapse = " and "), " variables")
+    variable.type.or <- paste0(paste0(variable.type, collapse = " or "), " variables")
+    net.txt.2 <- ngettext(length(nets), "this NET and its hidden", "these NETs and their hidden")
+    warning("NETs are removed from this analysis unless all their codes are not observed elsewhere. ",
+            "The ", variable.type.and, " have ", net.txt, net.name.txt, ", with both hidden and observed codes. ",
```

Comment:
where is the "both hidden and observed codes" portion of this warning coming from?

Reply:
Updated/rephrased in https://github.com/Displayr/flipRegression/commit/efac6e9578e3e4e9fda3f18d26bfdb84ce13779c.